### DISCUSSION TITLE
Allow a way to fetch information for the whole app

### DISCUSSION THREAD

**Discussion by kud** [vercel/next.js#21885] [Ideas]
Hello,

I read the doc and seems that `_app_ ` can't have any function to fetch some information.

It could be great to allow next.js to be able to fetch some information for the whole application like for the layout (footer, header, etc).

I've got some dynamic pages from a dashboard, and I wanted to display their links into my footer.

The footer is inside every page, and I don't want to code this for every page, I'd like to be able to do it just one time for the whole application.

**Comment by cobrexus** [comment]
The [Custom App](https://nextjs.org/docs/advanced-features/custom-app) feature in the Next.js docs is created using `/pages/_app.js`, and it *does* allow for data fetching, using `getInitialProps`, which will make a server request on initial page load, and do client-side fetching for each subsequent navigation.

Check out the example:

```js
function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

// Only uncomment this method if you have blocking data requirements for
// every single page in your application. This disables the ability to
// perform automatic static optimization, causing every page in your app to
// be server-side rendered.
//
// MyApp.getInitialProps = async (appContext) => {
//   // calls page's `getInitialProps` and fills `appProps.pageProps`
//   const appProps = await App.getInitialProps(appContext);
//
//   return { ...appProps }
// }

export default MyApp
```

**Reply by kud** [reply]
Yes, but I was stuck with this line:

> App currently does not support Next.js Data Fetching methods like getStaticProps or getServerSideProps.

**Reply by cobrexus** [reply]
@kud `getStaticProps` and `getServerSideProps` don't work in a custom app, you have to use `getInitialProps`.
